realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1428,12 +1441,92 @@ public long count() {
      * @throws java.lang.RuntimeException Any other error
      */
     public RealmResults<E> findAll() {
+        checkQueryIsNotReused();
         return new RealmResults<E>(realm, query.findAll(), clazz);
     }
 
     /**
      * Find all objects that fulfill the query conditions and sorted by specific field name.
+     * (Need a Realm opened from a looper thread to work).
```

Comment:
I'm not a big fan of putting description in `()`. Also perhaps `This method is only available from a Looper thread`?